File: 2026POTD/JANUARY/03-01-2026/1411.php
class Solution {

    /**
     * @param Integer $n
     * @return Integer
     */
    function numOfWays($n) {
        $MOD = 1000000007;
        $aba = 6;
        $abc = 6;
        for ($i = 2; $i <= $n; $i++) {
            $newAba = ($aba * 3 + $abc * 2) % $MOD;
            $newAbc = ($aba * 2 + $abc * 2) % $MOD;
            $aba = $newAba;
            $abc = $newAbc;
        }
        return ($aba + $abc) % $MOD;
    }
}

File: 2026POTD/JANUARY/03-01-2026/1411.c
int numOfWays(int n) {
    const int MOD = 1000000007;
    long long aba = 6, abc = 6;
    for (int i = 2; i <= n; i++) {
        long long newAba = (aba * 3 + abc * 2) % MOD;
        long long newAbc = (aba * 2 + abc * 2) % MOD;
        aba = newAba;
        abc = newAbc;
    }
    return (int)((aba + abc) % MOD);
}